Doc/library/doctest.rst

```diff
@@ -300,33 +301,49 @@ their contained methods and nested classes.
 How are Docstring Examples Recognized?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+The :mod:`doctest` module treats any line beginning with ``>>>`` as an 
+example to be tested. 
+Following lines which begin with ``...`` continue the example. 
+Subsequent non-blank lines, if any, form an expected output string. 
+A blank (all-whitespace) line, or a line with ``>>>`` (beginning the 
+next example), ends the expected output string.
+
 In most cases a copy-and-paste of an interactive console session works fine,
 but doctest isn't trying to do an exact emulation of any specific Python shell.
 
 ::
 
    >>> # comments are ignored
-   >>> x = 12
-   >>> x
-   12
-   >>> if x == 13:
-   ...     print("yes")
+   ... 
+   >>> import math
+   >>> x = factorial(10); math.ceil(math.log10(x))
+   7
+   >>> if x == factorial(9)*10:
+   ...     print("same:\n{0}".format(x))
    ... else:
-   ...     print("no")
-   ...     print("NO")
-   ...     print("NO!!!")
+   ...     print("differ:\n{0}\n{1}".format(x, factorial(9)*10))
```

Comment:
@JulienPalard , thank you for your review. 
The core of this PR is that I wrote an "example" which required an `import` statement to run, and this documentation didn't explain clearly to me how I could do this right. So, for me as a reader, the existing text was not sufficient:
```
>>> x = 12
>>> x
12
```
And the core of what I am proposing is to use a different demonstration of multiple "examples" which build up state and then use it, as in:
```
>>> import math
>>> x = factorial(10)
>>> math.ceil(math.log10(x))
7
```
(I believe there is a benefit of using a statement group with `;` instead of simple statements, because it obliquely explains one more aspect of how the doctest feature works, but that is a side issue.)

You said, "Please do not change the example, it was simple and covered the point you're trying to fix."  My feedback to you is that, as a developer and a reader of this documentation, it did _not_ cover the point. It did not tell me how to use use an `import` statement in my "example". The core of this PR is to make documentation do a better job of covering this point.